js/client-mainmenu.js

```diff
@@ -1122,12 +1122,60 @@
 					for (var i = 0; i < teams.length; i++) {
 						if ((!teams[i].format && !teamFormat) || teams[i].format === teamFormat) {
 							var selected = (i === curTeam);
-							bufs[curBuf] += '<li><button name="selectTeam" value="' + i + '"' + (selected ? ' class="sel"' : '') + '>' + BattleLog.escapeHTML(teams[i].name) + '</button></li>';
-							count++;
-							if (count % bufBoundary == 0 && curBuf < 4) curBuf++;
+							if (this.folderToggleOn !== true) {
+								bufs[curBuf] += '<li><button name="selectTeam" value="' + i + '"' + (selected ? ' class="sel"' : '') + '>' + BattleLog.escapeHTML(teams[i].name) + '</button></li>';
+								count++;
+								if (count % bufBoundary == 0 && curBuf < 4) curBuf++;
+							} else {
+								var folderName = teams[i].folder || "";
+								if (folderName !== "") {
+									if (folders[folderName] === undefined) folders[folderName] = []; // create empty array to push to
```

Comment:
similarly, you can just do an `if (folders[folderName])` check. theres nothing *wrong* per se with being super explicit, but its just added verbosity and makes the reader question 'why is the code so explicit here, could `folders[folderName]` really be falsey and not undefined?'

Reply:
This is the call inside, and folders is the container for the folders. This starts out as an empty json object. When I first initialize that parameter of folderName, I want to create an array. So I'm declaring as an empty array when it's undefined. I'm not being needlessly explicit here, but is there a better way to write that?

Reply:
`if (folders[foldersName])` will work if  `folders[foldersName] === undefined`. `[]` is not falsey.